src/nvim/option.c

```diff
@@ -1359,6 +1359,10 @@ do_set (
             && nextchar != NUL && !ascii_iswhite(afterchar))
           errmsg = e_trailing;
       } else {
+
+        int value_is_replaced = !prepending && !adding && !removing;
```

Comment:
In the original patch, this line already exists. Thus this patch is based on another earlier patch that should be merged first.

Reply:
https://github.com/vim/vim/commit/247bb7e43b47eb8407a1111bed60b61aceda52ad

Reply:
Shall I include it in a separate request or just merge it here?

Reply:
Here, please.

Reply:
I found out we'll need a bunch of other patches before this one as well, some of which are not directly related to this issue, so I'm temporarily keeping this one aside and starting with the older ones...

Reply:
Yeah, it's hard to keep up with porting patches while developing Nvim at the same time. So, any help there is more than welcome.

Maybe you used it for this commit already, but in case you didn't read it so far: https://github.com/neovim/neovim/wiki/Merging-patches-from-upstream-vim